src/main/java/com/github/quiltservertools/ledger/mixin/EndCrystalEntityMixin.java

```diff
@@ -24,9 +25,11 @@ public PlayerEntity getCausingPlayer() {
 
     @ModifyArgs(method = "damage", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/World;createExplosion(Lnet/minecraft/entity/Entity;DDDFLnet/minecraft/world/explosion/Explosion$DestructionType;)Lnet/minecraft/world/explosion/Explosion;"))
     public void correctEndCrystalEntitySource(Args args, DamageSource source, float amount) {
-        if (source.getSource() instanceof PlayerEntity player) {
-            this.causingPlayer = player;
-            args.set(0, (EndCrystalEntity) (Object) this);
+        if (source.getSource() instanceof PlayerEntity) {
```

Comment:
Please use `source.getSource() instanceof PlayerEntity player` instead of manually casting